packages/api/amplify_api_dart/lib/src/api_plugin_impl.dart

```diff
@@ -96,10 +120,13 @@ class AmplifyAPIDart extends APIPluginInterface with AWSDebuggable {
   ///
   /// Register OIDC/Lambda set to _authProviders in constructor.
   void _registerApiPluginAuthProviders() {
-    _apiConfig.endpoints.forEach((key, value) {
+    _dataConfig?.forEach((key, value) {
       // Check the presence of apiKey (not auth type) because other modes might
       // have a key if not the primary auth mode.
-      if (value.apiKey != null) {
+      if (value.apiKey != null ||
```

Comment:
Q: Do you need `value.apiKey != null`? Is there ever a scenario where this would be true while the other half of the statement is false?

Reply:
Initially, I removed it thinking the same thing, then I reconsidered after reading the code comment and was cautious about deleting it.

Reply:
The comment doesn't make sense to me anymore now that you are also checking the other auth modes. `value.apiKey != null` was a hack since checking the default auth mode would not have been enough since a non default auth mode could be API key.

We should not add redundant code out of precaution. If there is a use case where apiKey will be present but non of the auth modes will be `apiKey` (or vice versa) then we likely need to handle that in other locations as well.